model-optimizer/extensions/middle/EltwiseInputReshape.py

```diff
@@ -65,58 +65,48 @@ def find_and_replace_pattern(self, graph: Graph):
 
 
 class EltwiseInputReshape(MiddleReplacementPattern):
-    enabled = True
-    force_clean_up = True
-
-    def run_after(self):
-        from extensions.middle.pass_separator import MiddleStart
-        return [MiddleStart]
+    # This pass should be called directly from pipeline before layout change and other permutations
+    enabled = False
 
     def find_and_replace_pattern(self, graph: Graph):
-        for node in graph.get_data_nodes():
-            # Get all requested shapes for current node
-            # This mapping will contain pairs like {shape:[list of consumers nodes]}
-            mapping = {}
-            for consumer in node.out_nodes():
-                edge_attrs = graph.get_edge_data(node.id, consumer.id)[0]
-                if 'new_shape' in edge_attrs:
-                    if np.array_equal(edge_attrs['new_shape'], node.shape):
-                        continue
-                    new_shape = tuple([x for x in edge_attrs['new_shape']])
-                    if not new_shape in mapping:
-                        mapping.update({new_shape: [consumer]})
-                    else:
-                        mapping[new_shape].append(consumer)
+        for node in graph.get_op_nodes():
```

Comment:
It is much more efficient to iterate over all `is_eltwise` operations here and check their input ports shapes and output port shape. Currently you iterate over all graph nodes, then for each output port of the operation you get the consumers and if the consumer is eltwise operation do the magic. Very inefficient.

Reply:
Fixed. As we agreed now it is two pass transformation: generate a map for eltwise producers with non-normalized shapes; walkthrough the map and insert reshape.